File: projects/order-creation/core/src/main/java/com/robinfood/core/entities/PortionEntity.java
package com.robinfood.core.entities;

import java.math.BigDecimal;
import lombok.Data;

@Data
public class PortionEntity {

    private final ReplacementPortionEntity changedTo;

    private final Integer free;

    private final Long id;

    private final Boolean included;

    private final String name;

    private final BigDecimal price;

    private final Long unitNumber;

    private final Integer quantity;

    private final String sku;

}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/entities/transactionrequest/PortionEntity.kt
package com.robinfood.core.entities.transactionrequest

import java.math.BigDecimal

data class PortionEntity(
    val discount: BigDecimal,
    val free: Int,
    val id: Long,
    val isIncluded: Boolean,
    val name: String,
    val price: BigDecimal,
    val product: PortionProductEntity,
    val quantity: Int,
    val replacementPortion: ReplacementPortionEntity?,
    val sku: String,
    val unitId: Long,
    val unitNumber: Double
)
